sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -62,65 +62,68 @@ object SQLExecution {
    * we can connect them with an execution.
    */
   def withNewExecutionId[T](
-      sparkSession: SparkSession,
       queryExecution: QueryExecution,
       name: Option[String] = None)(body: => T): T = {
```

Comment:
can we do `...: T = queryExecution.sparkSession.withActive {` here? This can save a lot of diff.